include/malloc.h

```diff
@@ -36,6 +38,9 @@ typedef struct malloc_pool {
   uint32_t mp_magic;                /* Detect programmer error. */
   const char *mp_desc;              /* Printable type name. */
   struct ma_list mp_arena;          /* Queue of managed arenas. */
+  mtx_t mutex;                      /* Mutex protecting structure */
```

Comment:
Please use common prefix for structure fields, i.e. `mp_`.

Fields naming could also be made more reasonable:
* `mutex` -> `mp_lock`
* `pages_used` -> `mp_pages_min`
* `pages_limit` -> `mp_pages_max`

Actual number of used pages should somewhere between `min` and `max` values.